### PR TITLE
Feature/comments and docs and fixes

### PR DESCRIPTION
**Pull request by GlynLeine**

### CODE REVIEW

**Comment by GlynLeine** [comment]
Everything is kind of random because i got off track a lot. right now these changes need to happen before this get's too big.

File: args/application/context/contexthelper.hpp [hunk 1]
```diff
@@ -26,21 +34,36 @@ namespace args::application
...
         ContextHelper() = delete;
         ~ContextHelper() = delete;
 
+        /**@brief Checks if the context has been initialized.
+         */
         static bool initialized();
         static bool init();
+
+        /**@brief Binds callback to be executed right after the context has been initialized. The callback will be invoked immediately if the context is already initialized.
+         */
```
Comment (Algo-ryth-mix) [inline args/application/context/contexthelper.hpp]: Long line, should probably be broken

File: args/application/events/windowevents.hpp [hunk 2]
```diff
@@ -27,6 +27,18 @@ namespace args::application
...
+        math::ivec2 position;
+        math::ivec2 size;
+
+        window_fullscreen_request(id_type entityId = invalid_id, math::ivec2 position = { 100 ,100 }, math::ivec2 size = { 400, 400 }) : entityId(entityId), position(position), size(size) {}
+
+        virtual bool persistent() override { return false; }
+        virtual bool unique() override { return false; }
+    };
+
     struct window_close final : public events::event<window_close>
```
Comment (Algo-ryth-mix) [inline args/application/events/windowevents.hpp]: kind of a bad name 
consider: `window_change_monitor_mode_request` or something similar (potentially something shorter)

File: args/application/window/window.hpp [hunk 3]
```diff
@@ -15,18 +15,35 @@ namespace args::application
...
+        bool isFullscreen;
         async::readonly_rw_spinlock* lock;
+        int swapInterval;
 
         operator GLFWwindow* () const { return handle; }
         window& operator=(GLFWwindow* ptr) { handle = ptr; return *this; }
 
-        inline void enableCursor(bool enabled)
+        inline void enableCursor(bool enabled) const
         {
```
Comment (Algo-ryth-mix) [inline args/application/window/window.hpp]: this requries some protection, considerations for future changes pls

File: args/application/window/windowsystem.hpp [hunk 4]
```diff
@@ -9,51 +9,59 @@
 
 namespace args::application
 {
+    /**@class WindowSystem
+     * @brief The system that's responsible for raising and polling all window events, swapping the buffers of the windows and creation and destruction of windows.
+     */
```
Comment (Algo-ryth-mix) [inline args/application/window/windowsystem.hpp]: long line, please break

File: args/application/window/windowsystem.hpp [hunk 5]
```diff
@@ -251,23 +286,69 @@ namespace args::application
...
+                    ContextHelper::setFramebufferSizeCallback(win, &WindowSystem::onWindowFrameBufferResize);
+                    ContextHelper::setWindowContentScaleCallback(win, &WindowSystem::onWindowContentRescale);
+                    ContextHelper::setDropCallback(win, &WindowSystem::onItemDroppedInWindow);
+                    ContextHelper::setCursorEnterCallback(win, &WindowSystem::onMouseEnterWindow);
+                    ContextHelper::setKeyCallback(win, &WindowSystem::onKeyInput);
+                    ContextHelper::setCharCallback(win, &WindowSystem::onCharInput);
+                    ContextHelper::setCursorPosCallback(win, &WindowSystem::onMouseMoved);
+                    ContextHelper::setMouseButtonCallback(win, &WindowSystem::onMouseButton);
+                    ContextHelper::setScrollCallback(win, &WindowSystem::onMouseScroll);
+                    ContextHelper::makeContextCurrent(nullptr);
```
Comment (Algo-ryth-mix) [inline args/application/window/windowsystem.hpp]: consider to make this piece of code a bit "dryer" the same definition is below again

File: args/core/logging/logging.hpp [hunk 6]
```diff
@@ -80,6 +121,50 @@ namespace fmt
...
+        template <typename FormatContext>
+        auto format(const args::core::math::vec2& p, FormatContext& ctx) {
+            // auto format(const point &p, FormatContext &ctx) -> decltype(ctx.out()) // c++11
+              // ctx.out() is an output iterator to write to.
+            return format_to(
+                ctx.out(),
+                presentation == 'f' ? "({:.1f}, {:.1f})" : "({:.1e}, {:.1e})",
+                p.x, p.y);
+        }
+    };
```
Comment (Algo-ryth-mix) [inline args/core/logging/logging.hpp]: this formatter makes no sense, it is a specialization for `ivec2` but the format accepts a `vec2`

File: args/rendering/data/shader.hpp [hunk 7]
```diff
@@ -159,40 +189,67 @@ namespace args::rendering
...
+         * @param type Data type of the components in the tensor.
+         * @param normalized Normalize the tensors before sending to VRAM.
+         * @param stride Amount of bytes in-between valid data chunks.
+         * @param pointer Amount of bytes until the first valid data chunk. 
+         */
+        void set_attribute_pointer(GLint size, GLenum type, GLboolean normalized, GLsizei stride, GLsizei pointer)
         {
             glEnableVertexAttribArray(m_location);
-            glVertexAttribPointer(m_location, size, type, normalized, stride, pointer);
+            glVertexAttribPointer(m_location, size, type, normalized, stride, (const GLvoid*)pointer);
```
Comment (Algo-ryth-mix) [inline args/rendering/data/shader.hpp]: consider using `reinterpret_cast<>` instead

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]

File: args/application/context/contexthelper.cpp [hunk 8]
```diff
@@ -59,6 +61,47 @@ namespace args::application
         return glfwGetPrimaryMonitor();
     }
 
+    GLFWmonitor* ContextHelper::getCurrentMonitor(GLFWwindow* window)
+    {
+        int nmonitors, i;
+        int wx, wy, ww, wh;
```
Comment (Jelled1st) [inline args/application/context/contexthelper.cpp:67]: What are the wx, wy, ww, wh? and what do they stand for?

**Review by Algo-ryth-mix — APPROVED** [review]

**Review by Jelled1st — APPROVED** [review]

**Pull request merged**